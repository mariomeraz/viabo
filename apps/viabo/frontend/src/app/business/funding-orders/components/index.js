export { default as CancelFundingOrder } from './CancelFundingOrder'
export { default as ConciliateModal } from './ConciliateModal'
export { default as FundingOrdersTable } from './FundingOrdersTable'
