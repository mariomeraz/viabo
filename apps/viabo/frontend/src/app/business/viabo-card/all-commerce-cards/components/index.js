export { default as CommerceCard } from './CommerceAssignedCard'
