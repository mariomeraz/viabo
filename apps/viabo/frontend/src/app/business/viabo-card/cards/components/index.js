export * from './CardDetails'
export * from './CardsSidebar'
