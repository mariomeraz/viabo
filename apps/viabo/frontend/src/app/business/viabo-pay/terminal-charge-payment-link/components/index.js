export * from './ChargePaymentForm'
export * from './ChargePaymentLinkDetails'
