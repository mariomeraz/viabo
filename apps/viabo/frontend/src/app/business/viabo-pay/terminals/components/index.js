export * from './TerminalDetails'
export * from './TerminalsDrawer'
export * from './terminals/TerminalItem'
export * from './terminals/TerminalsList'
export * from './terminals/TerminalsSearch'
