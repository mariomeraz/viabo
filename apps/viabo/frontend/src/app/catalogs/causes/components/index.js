export { default as NewCauseDrawer } from './NewCauseDrawer'
export * from './NewCauseForm'
