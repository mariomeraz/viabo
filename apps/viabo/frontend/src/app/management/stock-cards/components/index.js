export * from './StockCardSidebar'
export * from './StockCardTable'
