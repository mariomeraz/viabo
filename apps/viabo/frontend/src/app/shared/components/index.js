export * from './RightPanelStyle'
export * from './CardsList'
export * from './RightPanel'
export * from './AssignCardTableToolbar'
