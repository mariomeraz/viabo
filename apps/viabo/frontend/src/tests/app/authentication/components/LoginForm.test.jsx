import { cleanup, waitFor } from '@testing-library/react'
import { afterEach, beforeEach, describe, expect, test, vi } from 'vitest'

import { LoginForm } from '@/app/authentication/components'
import { render, screen, user } from '@/tests/testUtils'

const mockLogin = vi.fn()
let mockError = null
let wrapper

vi.mock('../../../hooks', () => ({
  useSignIn: () => ({
    isLoading: false,
    mutate: mockLogin,
    error: mockError,
    isSuccess: Boolean(!mockError),
    setCustomError: vi.fn()
  })
}))

describe('Formulario de Inicio de Sesión', () => {
  beforeEach(() => {
    vi.clearAllMocks()
  })

  afterEach(() => {
    cleanup()
    mockError = null
  })

  test('Debería mostrar el formulario de inicio de sesión', () => {
    wrapper = render(<LoginForm />)
    const userInput = screen.getByLabelText(/Email/i)
    const passwordInput = screen.getByLabelText(/Contraseña/i)
    const button = screen.getByRole('button', { name: /Accesar a mi cuenta/i })

    expect(wrapper.container).toMatchSnapshot()
    expect(userInput).toBeInTheDocument()
    expect(passwordInput).toBeInTheDocument()
    expect(button).toBeInTheDocument()
  })

  test('Debería enviarse el formulario con la información de inicio de sesión', async () => {
    wrapper = render(<LoginForm />)
    const userInput = screen.getByLabelText(/Email/i)
    const passwordInput = screen.getByLabelText(/Contraseña/i)
    const button = screen.getByRole('button', { name: /Accesar a mi cuenta/i })

    await user.type(userInput, 'test@test.com')
    await user.type(passwordInput, 'test123')
    await user.click(button)

    await waitFor(() => expect(mockLogin).toHaveBeenCalledTimes(1))
  })

  test('Debería enviarse el formulario con la información de inicio de sesión y mostrar alerta de error', async () => {
    wrapper = render(<LoginForm />)
    const userInput = screen.getByLabelText(/Email/i)
    const passwordInput = screen.getByLabelText(/Contraseña/i)
    const button = screen.getByRole('button', { name: /Accesar a mi cuenta/i })
    mockError = true
    const scrollIntoViewMock = vi.fn()
    window.HTMLElement.prototype.scrollIntoView = scrollIntoViewMock

    await user.type(userInput, 'test@test.com')
    await user.type(passwordInput, 'test123')
    await user.click(button)

    await waitFor(() => {
      const alert = screen.getByRole('alert')
      expect(alert).toBeInTheDocument()
    })
  })
})
