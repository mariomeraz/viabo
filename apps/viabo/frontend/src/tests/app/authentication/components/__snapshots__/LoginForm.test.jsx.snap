// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Formulario de Inicio de Sesión > Debería mostrar el formulario de inicio de sesión 1`] = `
<div>
  <div
    class="Toastify"
  />
  <div
    class="MuiBox-root css-0"
  >
    <div
      class="MuiStack-root css-4lvoef-MuiStack-root"
    >
      <div
        class="MuiStack-root css-nen11g-MuiStack-root"
      >
        <div>
          <h3
            class="MuiTypography-root MuiTypography-h3 MuiTypography-alignCenter css-1a9rpoj-MuiTypography-root"
          >
            Aquí Comienza
          </h3>
        </div>
        <div>
          <h3
            class="MuiTypography-root MuiTypography-h3 MuiTypography-alignCenter css-17i1ark-MuiTypography-root"
          >
            tu agilidad en pagos
          </h3>
        </div>
      </div>
      <div
        style="min-height: 15px; opacity: 0;"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6 MuiTypography-alignCenter css-1au37kk-MuiTypography-root"
        />
      </div>
      <div
        style="opacity: 0;"
      >
        <form
          action="#"
        >
          <div
            class="MuiStack-root css-ib3th8-MuiStack-root"
          >
            <div
              class="MuiStack-root css-jfdv4h-MuiStack-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-animated css-1cv7szy-MuiFormLabel-root-MuiInputLabel-root"
                for="email"
              >
                Email
              </label>
              <div
                class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-10k2l18-MuiFormControl-root-MuiTextField-root"
              >
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-1622mpx-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input css-fw5cm8-MuiInputBase-input-MuiOutlinedInput-input"
                    id="email"
                    name="email"
                    placeholder="usuario@dominio.com"
                    type="email"
                    value=""
                  />
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-9425fu-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-ihdtdm"
                    >
                      <span
                        class="notranslate"
                      >
                        ​
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
            <div
              class="MuiStack-root css-jfdv4h-MuiStack-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-animated css-1cv7szy-MuiFormLabel-root-MuiInputLabel-root"
                for="password"
              >
                Contraseña
              </label>
              <div
                class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
              >
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd css-56wmv2-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-qhnzi4-MuiInputBase-input-MuiOutlinedInput-input"
                    id="password"
                    name="password"
                    placeholder="********"
                    type="password"
                    value=""
                  />
                  <div
                    class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-21twve-MuiInputAdornment-root"
                  >
                    <button
                      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeSmall css-1w1roue-MuiButtonBase-root-MuiIconButton-root"
                      tabindex="0"
                      type="button"
                    >
                      <svg
                        fill="currentColor"
                        height="1em"
                        stroke="currentColor"
                        stroke-width="0"
                        viewBox="0 0 256 256"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M234.42,162A12,12,0,1,1,213.58,174l-16.86-29.5a127.19,127.19,0,0,1-30.17,13.86L171.84,190a12,12,0,0,1-9.87,13.8,11.22,11.22,0,0,1-2,.17,12,12,0,0,1-11.82-10L143,163.17a136.5,136.5,0,0,1-30.06,0L107.84,194A12,12,0,0,1,96,204a11.22,11.22,0,0,1-2-.17A12,12,0,0,1,84.16,190l5.29-31.72a127.19,127.19,0,0,1-30.17-13.86L42.42,174A12,12,0,1,1,21.58,162L40,129.85a159.73,159.73,0,0,1-17.31-18.31A12,12,0,0,1,41.34,96.46C57.38,116.32,85.44,140,128,140s70.62-23.68,86.66-43.54a12,12,0,0,1,18.67,15.08A159.73,159.73,0,0,1,216,129.85Z"
                        />
                      </svg>
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-9425fu-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-ihdtdm"
                    >
                      <span
                        class="notranslate"
                      >
                        ​
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
            <div
              class="MuiStack-root css-1v3364e-MuiStack-root"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiLoadingButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-fullWidth MuiButton-root MuiLoadingButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-fullWidth css-1c0xctp-MuiButtonBase-root-MuiButton-root-MuiLoadingButton-root"
                id=":r3:"
                tabindex="0"
                type="submit"
              >
                Accesar a mi cuenta
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
            <p
              class="MuiTypography-root MuiTypography-body2 MuiTypography-alignCenter css-shwflf-MuiTypography-root"
            >
              ¿No tienes una cuenta?  
              <a
                class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-1xk92bt-MuiTypography-root-MuiLink-root"
                href="/registro"
              >
                Inscribete.
              </a>
            </p>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
